src/pages/scientificServices/pipelines/components/inputs/PipelineFileInput.tsx

```diff
@@ -23,33 +23,65 @@ interface PipelineInputSelectorProps {
   selectedFile: File | null;
   uploadState?: PipelineInputFileUploadState;
   onFileSelect: (file: File | null) => void;
+  onValidation(error?: string): void;
+  validationError?: string;
   onUploadComplete?: () => void;
-  setUploadState?: React.Dispatch<React.SetStateAction<Record<string, PipelineInputFileUploadState>>>;
+  setUploadState?: Dispatch<SetStateAction<Record<string, PipelineInputFileUploadState>>>;
 }
 
 export const PipelineFileInput: React.FC<PipelineInputSelectorProps> = ({
   input,
   selectedFile,
   uploadState,
   onFileSelect,
+  onValidation,
+  validationError,
   onUploadComplete,
   setUploadState,
 }) => {
   const fileInputRef = useRef<HTMLInputElement>(null);
-  const isFileValid = selectedFile && selectedFile.name.endsWith(input.fileSuffix || '');
-  const { label } = INPUT_DESCRIPTIONS[input.name];
+  const { label, validationRegex } = INPUT_DESCRIPTIONS[input.name];
+
+  const validateFile = (file: File | null) => {
+    // Check for required file
+    if (input.isRequired && !file) {
+      onValidation('This file is required.');
+      return;
+    }
+
+    // Check for valid file type
+    // N.B. this suffix is supplied separately by the backend, which is why it's not included in the validationRegex
+    if (file && input.fileSuffix && !file.name.endsWith(input.fileSuffix)) {
+      onValidation(`Invalid file type. Please upload a ${input.fileSuffix} file.`);
+      return;
+    }
+
+    // Check for valid file name using validation expression
+    if (file && validationRegex) {
+      const regex = new RegExp(validationRegex);
+      if (regex.test(file.name)) {
+        onValidation(undefined);
```

Comment:
just another noob question, is undefined like `null` here?  And setting the value to this just means like "passed validation"  when checked against like on line 90 in this file?